lua/nvim-treesitter/textobjects/move.lua

```diff
@@ -116,6 +261,30 @@ M.commands = {
       "-complete=custom,nvim_treesitter_textobjects#available_textobjects",
     },
   },
```

Comment:
for stuff like this it would be useful to add unit tests. Similar to the ones added here for a bug in swap nodes

https://github.com/nvim-treesitter/nvim-treesitter/blob/51d220e0bd8922caabcb3e5de936fc599a1bd210/tests/unit/ts_utils_spec.lua#L110-L148

You could quickly test different combinations and if there will be ever a bug with the current implementation. The PR could easily explain what went wrong. Plenary tests don't require a lot of code to set them up. Just set example test and apply the functions that are just added.

We can defer plenary tests to a different tests, but I think that they tackle a different problem that your Python snapshot tests.

Reply:
My python tests can generate snapshots, but it doesn't mean that's the only option. We can just manually create the yaml file that describe the unit test. If you don't mind I can add the builtin movements to that python tests as we have a good starting point.

If you think it's still better to have a plenary test in the repo, is it possible for you to create a basic template because you're more experienced with that?

Reply:
The advantage of plenary is that you don't need to do any setup. Just run `:PlenaryBustedFile %` to run the currently opened test file. For CI you just need to run a headless nvim run `-c ":PlenaryBustedDirectory`. If you're trying to do the same thing in Python you would need to implement all the logic they have in plenary. Most contributors would know how they work because Neovim also uses busted and many plugins use plenary for testing.

Just create a file `foo_spec.lua` with
```lua
describe("foo suite", function()
  it("can succeed", function()
    assert(1 + 1 == tonumber(vim.fn.eval("1 + 1")))
  end)
  it("it can fail", function()
    assert(1 + 1 == tonumber(vim.fn.eval("1 + 2")))
  end)
end)
```
is enough for a test suite. We don't need to install any dependency apart from the plenary plugin to run tests in Neovim. So I think adding plenary in addition to your Python tests wouldn't add to much bloat and overhead. Testing Lua functions directly in process is just more convent than to serialize and de-serialize over rtp. I'm talking about those kind of sanity checks that you would execute on two line examples. Python might be a better choice when you want to test whole repos or automated test where you only place example files in the repo.

But let's leave that for a another PR.

Reply:
I also agree that learning something standardised and native to the program is better for everyone. Yeah let's have both plenary and python tests running, and they can both test different things. Python can test more of actual functionality by using feedkeys and plenary can test actual function implementations. I will take a look at it when I have time. Thank you!